### PR TITLE
Fix issue with serialization(missing details); Fix msft health check …

### LINKED ISSUE #517: Health endpoint not reporting status and description of Microsoft health checks
## Describe the bug
Since upgrading to Steeltoe 3.0, status and description of individual Microsoft `IHealthCheck` interfaced health checks is not reported from the health endpoint.

With Steeltoe 2.4.4 `GET /actuator/health` returned:

```json
{
    "status": "UP",
    "diskSpace": {
        "total": 107371753472,
        "free": 69270450176,
        "threshold": 10485760,
        "status": "UP"
    },
    "Custom Service": {
        "status": "UP",
        "description": "Awesome service instance description"
    }
}
```

With Steeltoe 3.0.1 `GET /actuator/health` returns:

```json
{
    "status": "UP",
    "liveness": {},
    "diskSpace": {
        "total": 107371753472,
        "free": 69270450176,
        "threshold": 10485760,
        "status": "UP"
    },
    "readiness": {},
    "Custom Service": {}
}
```

## Steps to reproduce 
1. Setup Steeltoe Health Actuator
2. Implement a health check from `Microsoft.Extensions.Diagnostics.HealthChecks.IHealthCheck`
3. Return `Task.FromResult(new HealthCheckResult(HealthStatus.Healthy, "Awesome service instance description"))` from implemented `CheckHealthAsync` method
4. Register health check in Startup.cs using `services.AddHealthChecks().AddCheck<MyCustomHealthCheck>();`

## Expected behavior
Status and description of the health check should show up in health report when calling `GET /actuator/health`.

## Environment (please complete the following information):
 - Steeltoe Version [e.g. 3.0.1]
 - Platform: Cloud Foundry 
 - OS: Linux
 - .NET Version: .NET Core 3.1

### PR DESCRIPTION
**Pull request by hananiel**
…status and description fields

resolves #517 , #507

### CODE REVIEW

File: src/Management/test/EndpointCore.Test/Health/EndpointMiddlewareTest.cs [hunk 1]
```diff
@@ -163,7 +165,8 @@ public async void HealthActuator_ReturnsMicrosoftHealthDetails()
             var health = JsonSerializer.Deserialize<Dictionary<string, object>>(json);
             Assert.NotNull(health);
             Assert.True(health.ContainsKey("status"));
-            Assert.True(health.ContainsKey("diskSpace"));
+            Assert.True(health.ContainsKey("details"));
```
Comment (christianerbsmehl) [inline src/Management/test/EndpointCore.Test/Health/EndpointMiddlewareTest.cs:168]: Comment in line 164 is lying now 😉 Structure should be the following:
`{ "status":"UP", "details": { "diskSpace": { "total":499581448192, "free":407577710592, "threshold":10485760, "status": "UP" } } }`

**Review by christianerbsmehl — COMMENTED** [review]
Reviewed your changes, looks good. Just found an outdated comment in a test. Should fix the linked issues (including mine), thanks.

File: src/Management/src/EndpointBase/Health/HealthConverter.cs [hunk 2]
```diff
@@ -26,10 +26,17 @@ public override void Write(Utf8JsonWriter writer, HealthEndpointResponse value,
                     writer.WriteString("description", health.Description);
                 }
 
-                foreach (var detail in health.Details)
+                if (health.Details != null && health.Details.Count > 0)
                 {
-                    writer.WritePropertyName(detail.Key);
-                    JsonSerializer.Serialize(writer, detail.Value, options);
+                    writer.WritePropertyName("details");
```
Comment (jkonicki) [inline src/Management/src/EndpointBase/Health/HealthConverter.cs:31]: Shouldn't this still be detail.Key?

Comment (hananiel) [reply src/Management/src/EndpointBase/Health/HealthConverter.cs:31]: No, that merge formatting is misleading.  We still serialize `detail.Key` as we iterate over details object, but now it has an extra nesting with the string "details".

**Review by TimHess — APPROVED** [review]

**Pull request merged**